src/parser.ts

```diff
@@ -453,6 +474,240 @@ class Parser {
+    const [baseWithoutQuery, ...queries] = url.value.split('?');
+
+    if (!url.value.startsWith('/')) {
+      throw syntaxError(this.lexer.source, this.lexer.token.start, `incorrect or missed url`);
+    }
+
+    const paths = baseWithoutQuery
+      .split('/')
+      .filter((str) => str.startsWith(':'))
+      .map((a) => a.substring(1))
+      .map((a) => a.split(':'));
+
+    const pathsNodes = paths.map(([path, type]) =>
+      this.node<EndpointParameterPathNode>(this.lexer.token, {
+        kind: ASTNodeKind.ENDPOINT_PARAMETER_PATH,
+        type: type
+          ? this.node<EndpointParameterPathTypeNode>(this.lexer.token, {
+              kind: ASTNodeKind.ENDPOINT_PARAMETER_PATH_TYPE,
+              name: this.node<NameNode>(this.lexer.token, {
+                kind: ASTNodeKind.NAME,
+                value: type,
+              }),
+            })
+          : undefined,
+        name: this.node<NameNode>(this.lexer.token, {
+          kind: ASTNodeKind.NAME,
+          value: path,
+        }),
+      }),
+    );
+
+    const queryNodes = queries.map((q) =>
+      this.node<EndpointsParameterQueryNode>(this.lexer.token, {
+        kind: ASTNodeKind.ENDPOINT_PARAMETER_QUERY,
+        name: this.node<NameNode>(this.lexer.token, {
+          kind: ASTNodeKind.NAME,
+          value: q,
+        }),
+      }),
+    );
+
+    return [
+      this.node<NameNode>(this.lexer.token, {
+        kind: ASTNodeKind.NAME,
+        value: baseWithoutQuery,
+      }),
+      [...queryNodes, ...pathsNodes].map((typeNode) =>
+        this.node<EndpointsParameterNode>(this.lexer.token, {
+          kind: ASTNodeKind.ENDPOINT_PARAMETER,
+          type: typeNode,
+        }),
+      ),
+    ];
+  }
+
+  parseUrlTypeDefinitionNode(): EndpointUrlNode {
+    const name = this.parseName();
+    const request = this.parseEndpointParameterRequestNode();
+    const [cleanedName, parameters] = this.parseUrlParametersNode(name);
+
+    const params = request ? [request, ...parameters] : parameters;
+
+    return this.node<EndpointUrlNode>(this.lexer.token, {
+      kind: ASTNodeKind.ENDPOINT_URL,
+      name: cleanedName,
+      parameters: params.length > 0 ? params : undefined,
+    });
+  }
+
+  parseEndpointNamespaceTypeDefinitionNode(): EndpointTypeDefinitionNode {
+    const optionalDescription = this.parseModelDescription();
+
+    const securityDefinition = this.parseSecurityDefinition();
+
+    if (this.peek(TokenKind.DOLLAR)) {
+      throw this.throwNoCrudl();
+    }
+    
+    const verb = this.parseEndpointVerb();
+    const url = this.parseUrlTypeDefinitionNode();
+    const responses = this.parseEndpointResponses();
+
+    return this.node<EndpointTypeDefinitionNode>(this.lexer.token, {
+      kind: ASTNodeKind.ENDPOINT_TYPE_DEFINITION,
+      verb,
+      description: optionalDescription,
+      securityDefinition,
+      url,
+      responses: responses ? responses : undefined,
```

Comment:
as I said earlier it's better to have an empty array.

It's all about v8 hidden classes, it's better to have one shape of object